src/CodeGen_OpenGL_Dev.cpp

```diff
@@ -225,6 +233,69 @@ string CodeGen_GLSLBase::print_type(Type type, AppendSpaceIfNeeded space_option)
     return oss.str();
 }
 
+// The following comparisons are defined for ivec and vec
+// types, so we don't use call_builtin
+void CodeGen_GLSLBase::visit(const EQ *op) {
+    if (op->type.is_vector()) {
+        Type bool_type = Bool(op->type.lanes());
+        print_expr(Call::make(bool_type, "equal", {op->a, op->b}, Call::Extern));
```

Comment:
op->type should be equal to bool_type, so you could just use that (saves a bit of repetitive code).